src/core/mount.c

```diff
@@ -1891,9 +1913,21 @@ static int mount_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 device_found_node(m, what, 0, DEVICE_FOUND_MOUNT);
         }
 
+        m->mount_last_read = this_run;
+        m->mount_last_duration = now(CLOCK_MONOTONIC) - this_run;
+
         return 0;
 }
 
+static int mount_dispatch_timeout_io(sd_event_source *source, usec_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        sd_event_source_set_enabled(m->mount_event_source, SD_EVENT_ON);
+        m->mount_timeout_source = sd_event_source_unref(source);
+        return mount_dispatch_io(NULL, mnt_monitor_get_fd(m->mount_monitor),
+                                 EPOLLIN, userdata);
```

Comment:
can't say i am fan of invoking IO handlers with a NULL source here. that's just ugly... can't we find a different way? maybe just move the processing of the /proc/self/mountinfo changes into a new call of its own, that gets clean arguments